core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -53,7 +53,7 @@ public void deleteFile(String path) {
     try {
       fs.delete(toDelete, false /* not recursive */);
     } catch (IOException e) {
-      throw new RuntimeIOException(e, "Failed to delete file: %s", path);
+      throw new RuntimeIOException("Cannot delete the file: %s", path, e.getMessage());
```

Comment:
Please revert this change.

Caught exceptions should almost always be added to the exception thrown, and the error message was fine.

This also breaks formatting because there are two arguments but only one `%s` in the format string.